pkg/standalone-utils/contracts/relayer/IBaseRelayerLibrary.sol

```diff
@@ -50,4 +53,74 @@ abstract contract IBaseRelayerLibrary is AssetHelpers {
     function _setChainedReferenceValue(uint256 ref, uint256 value) internal virtual;
 
     function _getChainedReferenceValue(uint256 ref) internal virtual returns (uint256);
+
+    /**
+     * @dev This reuses `_resolveAmountAndPullToken` to adjust the `amount` if it is a chained reference, and
+     * pulls that amount of the `mainToken` to the relayer. Additionally, approve the `wrappedToken` to enable
+     * wrapping operations.
+     */
+    function _resolveAmountPullAndApproveToken(
+        IERC20 mainToken,
+        IERC20 wrappedToken,
```

Comment:
Would suggest making this an `address` since that's all it really needs to be in order to make this approval, and in some cases (like Euler and Gearbox) we are writing things like this which don't make any sense because these aren't actually tokens:

```
IERC20(address(eulerProtocol))
```